test/sources/npm_test.rb

```diff
@@ -80,5 +80,27 @@
         end
       end
     end
+
+    describe "missing dependencies (glob is missing package)" do
+      let(:fixtures) { File.expand_path("../../fixtures/npm/missing-dependencies", __FILE__) }
+
+      it "includes missing dependencies when yarn.lock is present" do
+        Dir.chdir fixtures do
+          File.unlink("yarn.lock") if File.exist? "yarn.lock"
+
+          assert source.dependencies.detect { |dep| dep.name == "autoprefixer" }
+          assert source.dependencies.detect { |dep| dep.name == "glob" }
+        end
+      end
+
+      it "excludes missing dependencies when yarn.lock is present" do
+        Dir.chdir fixtures do
+          File.write("yarn.lock", "") unless File.exist? "yarn.lock"
```

Comment:
Alternatively if you want to follow my other suggestion about copying the fixtures to a tmp folder and then manipulating the tmp folder to get the correct state for the test, this doesn't need to be cleaned up
```suggestion
          File.write("yarn.lock", "")
```

Reply:
N/A after test rework.